### PR TITLE
enhance: add context to scm

### PR DESCRIPTION
**Pull request by plyr4**
continuing the efforts started in #898 #922 #923 #930 #937 #938 #941 #942 #950 #951

this is prep to add support for OpenTelemetry tracing.

the reason being, otel can (for the most part) work out of the box if you pass around contexts properly.

### CODE REVIEW

**Review by wass3rw3rk — APPROVED** [review]
context everywhere

**Review by wass3rw3rk — APPROVED** [review]

File: cmd/vela-server/schedule.go [hunk 1]
```diff
@@ -240,7 +240,7 @@ func processSchedule(ctx context.Context, s *library.Schedule, compiler compiler
 		pipeline, err = database.GetPipelineForRepo(context.TODO(), b.GetCommit(), r)
```
Comment (wass3rw3rk) [inline cmd/vela-server/schedule.go:240]: should this TODO?

**Review by wass3rw3rk — CHANGES_REQUESTED** [review]

File: cmd/vela-server/schedule.go (hunk 1 above)
Comment (plyr4) [reply cmd/vela-server/schedule.go:240]: nope, oversight from a previous PR good catch

**Review by wass3rw3rk — APPROVED** [review]

**Review by KellyMerrick — APPROVED** [review]
context.TODO(lgtm)

**Review by wass3rw3rk — APPROVED** [review]

**Pull request merged**